builtin/providers/aws/resource_aws_instance.go

```diff
@@ -365,16 +404,18 @@ func resourceAwsInstanceCreate(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return fmt.Errorf(
 			"Error waiting for instance (%s) to become ready: %s",
-			instance.InstanceId, err)
+			*instance.InstanceID, err)
 	}
 
 	instance = instanceRaw.(*ec2.Instance)
 
 	// Initialize the connection info
-	d.SetConnInfo(map[string]string{
-		"type": "ssh",
-		"host": instance.PublicIpAddress,
-	})
+	if instance.PublicIPAddress != nil {
+		d.SetConnInfo(map[string]string{
```

Comment:
What is this `setConnInfo` used for? Seems we were setting it before, even if there was no public IP?